.github/workflows/integration-tests.yml

```diff
@@ -0,0 +1,137 @@
+      with:
+        key: ${{ runner.os }}-v3-npm-${{ hashFiles('**/package-lock.json') }}
+        path: ~/.npm
+
+    - id: sdkman
+      if: steps.grails-sdk.outputs.cache-hit != 'true'
+      name: download grails SDK (unless cached)
+      uses: sdkman/sdkman-action@master
+      with:
+        candidate: grails
+        version: 1.3.9
+
+    - if: steps.grails-sdk.outputs.cache-hit != 'true'
+      name: unzip grails SDK (unless cached)
+      run: unzip -o ${{ steps.sdkman.outputs.file }} -d ~/
+
+    - name: npm install
+      run: npm install
+
+    - name: npm rebuild node-sass
+      run: npm rebuild node-sass
+
+    - name: npm run bundle
+      run: npm run bundle
+
+    - name: configure OpenBoxes for database access and testing
+      run: |
+        mkdir -p ~/.grails
+        echo "dataSource.password=${{ env.OPENBOXES_PASSWORD }}" > ~/.grails/openboxes-config.properties
+        echo "dataSource.url=jdbc:mysql://localhost:3306/${{ env.DB_DATABASE }}?allowPublicKeyRetrieval=true&useSSL=false" >> ~/.grails/openboxes-config.properties
+        echo "dataSource.username=${{ env.OPENBOXES_USER }}" >> ~/.grails/openboxes-config.properties
+        echo "openboxes.fixtures.enabled=true" >> ~/.grails/openboxes-config.properties
+        echo "openboxes.jobs.dataCleaningJob.enabled=false" >> ~/.grails/openboxes-config.properties
+        echo "openboxes.jobs.refreshDemandDataJob.enabled=false" >> ~/.grails/openboxes-config.properties
+        echo "openboxes.jobs.refreshInventorySnapshotJob.enabled=false" >> ~/.grails/openboxes-config.properties
+        #
+        # FIXME Strangely, enabled = false doesn't disable refreshProductAvailabilityJob,
+        # FIXME but setting delayInMilliseconds to a high value has the same practical effect.
+        #
+        echo "openboxes.jobs.refreshProductAvailabilityJob.enabled=false" >> ~/.grails/openboxes-config.properties
+        echo "openboxes.jobs.refreshProductAvailabilityJob.delayInMilliseconds=3600000" >> ~/.grails/openboxes-config.properties
+        echo "openboxes.refreshAnalyticsDataOnStartup.enabled=false" >> ~/.grails/openboxes-config.properties
+
+      #
+      # Liquibase migrations from scratch are very, very slow. For performance,
+      # this step pre-loads mysqldump output from a clean install of a recent
+      # OpenBoxes release. Technically, this step is optional in the sense that
+      # later steps would automatically run all the necessary migrations if it
+      # were removed. But those migrations would certainly take hours, and almost
+      # certainly fail with out-of-memory errors.
+      #
+    - name: preload empty database (v0.8.18-hotfix3)
+      run: |
+        gunzip -c ${{ github.workspace }}/test/data/preload.tgz | tar xvf -
+        mysql -p${{ env.DB_PASSWORD }} -u${{ env.DB_USER }} "${{ env.DB_DATABASE}}" < preload_schema.sql
+        mysql -p${{ env.DB_PASSWORD }} -u${{ env.DB_USER }} "${{ env.DB_DATABASE}}" < preload_data.sql
```

Comment:
^^^ @jmiranda by the time I read your comment on OBPIH-4608 I had already written this. Let me know if it's what you had in mind. (I generated these two files using the code in `OBPIH-4305-clone-remote-database`)

Reply:
If there's test data in preload_data.sql that is not in the Liquibase data migrations then this is probably fine and probably the preferred method as this would then be a test resource.

However if preload_schema.sql and preload_data.sql create the same database you'd get if you ran through the database migrations on an empty database, then I'd like for this to be more generic (it's no longer just a test resource). 

For example, for every new release (or after some other arbitrary milestone - every two months - it doesn't really matter) we should regenerate the baseline.sql and baseline-data.sql scripts (or just have one baseline.sql with everything). This would include the DATABASECHANGELOG table and include all of the database migrations that have been created up to this point. After installing this database, we start the application and the rest of the Liquibase migrations will be executed. And then the tests will be executed. 

The closer the baseline.sql is to the current data model, the less time will need to be spend running the migrations. But we could be on a very old baseline.sql and the outcome will be the same. The reason I want this to be generic is because it's not just applicable to testing. We have severs that would benefit from a clean build / deploy process as well and we could hook this script into that process.

With that said, we can keep this solution for now since it solves the immediate issue. But we should add a tech debt ticket to make this more generic.